doc/sphinx/source/inspection/debug_core.rst

```diff
@@ -0,0 +1,147 @@
+   ``NGRAPH_INTERPRETER_ENABLE``, Control the building of the ``INTERPRETER`` backend,  ``TRUE``
+   ``NGRAPH_INTERPRETER_STATIC_LIB_ENABLE``, Enable build INTERPRETER backend static library, ``FALSE``
+   ``NGRAPH_JSON_ENABLE``, Enable JSON based serialization and tracing features, ``TRUE``
+   ``NGRAPH_LIB_VERSIONING_ENABLE``  Enable shared library versioning, ``FALSE``
+   ``NGRAPH_MLIR_ENABLE``, Control the building of MLIR backend, ``FALSE``
+   ``NGRAPH_NOP_ENABLE``,  Control the building of the NOP backend,  ``TRUE``
+   ``NGRAPH_ONNX_IMPORT_ENABLE``  Enable ONNX importer, ``FALSE``
+   ``NGRAPH_PLAIDML_ENABLE``, Enable the PlaidML backend,  ``${PLAIDML_FOUND}``
+   ``NGRAPH_PYTHON_BUILD_ENABLE``,  Enable build of ``NGRAPH`` python package wheel, ``FALSE``
+   ``NGRAPH_STATIC_LIB_ENABLE``, Enable build ``NGRAPH`` static library, ``FALSE``
+   ``NGRAPH_TBB_ENABLE``, Only if (``NGRAPH_CPU_ENABLE``) Control usage of TBB for CPU backend, ``TRUE``
+   ``NGRAPH_TOOLS_ENABLE``  Control the building of tools, ``TRUE``
+   ``NGRAPH_UNIT_TEST_ENABLE``,  Control the building of unit tests, ``TRUE``
+   ``NGRAPH_USE_PREBUILT_MLIR``, Use the `precompiled MLIR`_,``FALSE``
+   ``NGRAPH_USE_PREBUILT_LLVM``, Use a precompiled LLVM  ,``FALSE``
+
+
+Environment Variables
+=====================
+
+.. important:: Many of the following flags may be experimental only and subject to change.
+
+
+.. csv-table::
+   :header: "Environment Variable", "Description"
+   :widths: 20, 35
+   :escape: ~
+
+   ``NGRAPH_CPU_BIN_TRACER_LOG``, See also :ref:`debug_tracer`
+   ``NGRAPH_CPU_DEBUG_TRACER``,  See also :ref:`debug_tracer`
+   ``NGRAPH_CPU_TRACER_LOG``, See also :ref:`debug_tracer`
+   ``NGRAPH_CPU_TRACING``, Generate Timelines for CPU to check in ``chrome://tracing``
+   ``NGRAPH_DECONV_FUSE``,  Default ``FALSE``; when ``TRUE`` it enables fusion for deconvolution.  Only available with CPU.
+   ``NGRAPH_ENABLE_SERIALIZE_TRACING``,  Creates serialized files to be run with ``nbench`` for localized execution rather than whole stack execution
+   ``NGRAPH_INTER_OP_PARALLELISM``, See also :ref:`interop_intraop`
+   ``NGRAPH_INTRA_OP_PARALLELISM``, See also :ref:`interop_intraop`
+   ``NGRAPH_PASS_ATTRIBUTES``, Enable or disable attributes related to a pass; see also `pass config`_
+   ``NGRAPH_PASS_ENABLES``,  Enable or disable a pass: either core or backend
+   ``NGRAPH_PROFILE_PASS_ENABLE``,  Per-pass time taken to compile
+   ``NGRAPH_VISUALIZE_EDGE_JUMP_DISTANCE``,  Calculated in code; helps prevent *long* edges between two nodes very far apart
+   ``NGRAPH_VISUALIZE_EDGE_LABELS``, Set it to 1 in ``~/.bashrc``
+   ``NGRAPH_VISUALIZE_TRACING_FORMAT``, Default format is ``.svg``
+   ``NGRAPH_VISUALIZE_TREE_OUTPUT_SHAPES``,  Set it to 1 in ``~/.bashrc``
+   ``NGRAPH_VISUALIZE_TREE_OUTPUT_TYPES``, Set it to 1 in ``~/.bashrc``
+   ``OMP_NUM_THREADS``, See `OpenMPI Runtime Library Documentation`_
+
```

Comment:
This list seems rather small than original and missing some important variables like NGRAPH_ENABLE_VISUALIZE_TRACING, NGRAPH_ENABLE_TRACING. Just wondering what happened.

Reply:
In the case of the flag being self-descriptive, it still needs an entry as a description so the CSV formatting we're using makes sense.

So to add a new item just specify both the exact flag and the description; any missing flags would just need a longer description.

Reply:
We have an entire documentation section on those flags already; added the link to them too"

Reply:
I scrubbed the list and narrowed it down to these core env vars
```

NGRAPH_DISABLE_LOGGING | Disable printing all logs irrespective of build type
-- | --
NGRAPH_DISABLED_FUSIONS | Disable specified fusions. Specified as ; separated list and supports regex
NGRAPH_ENABLE_REPLACE_CHECK | Enables strict type checking in copy constructor copy_with_new_args
NGRAPH_ENABLE_SERIALIZE_TRACING | Enables creating serialized json files to be run with nbench. Generates 1 file per pass
NGRAPH_ENABLE_TRACING | Enables creating graph execution timelines to be viewed in chrome://tracing
NGRAPH_ENABLE_VISUALIZE_TRACING | Enables creating visual graph for each pass. By default generates .svg files
NGRAPH_FAIL_MATCH_AT | Allows one to specify node name patterns to abort pattern matching at particular nodes. Helps debug an offending fusion
NGRAPH_GTEST_INFO | Enables printing info about a specific test
NGRAPH_INTER_OP_PARALLELISM | See the reference
NGRAPH_INTRA_OP_PARALLELISM | See the reference
NGRAPH_PASS_ATTRIBUTES | Specify pass specific attributes to be enabled or disabled. specify a semi-colon separated list. Naming of pass attributes is up to the backends.
NGRAPH_PASS_ENABLES | Specify a semi-colon separated list to enable or disable a pass. This will override the default enable/disable values
NGRAPH_PROFILE_PASS_ENABLE | Dump the name and execution time of each pass
NGRAPH_PROVENANCE_ENABLE | Enable adding provenance info to nodes. This will also be added to serialized files.
NGRAPH_SERIALIZER_OUTPUT_SHAPES | Enable adding output shapes in the serialized graph
NGRAPH_VISUALIZE_EDGE_JUMP_DISTANCE | Calculated in code; helps prevent *long* edges between two nodes very far apart
NGRAPH_VISUALIZE_EDGE_LABELS | When enabled, adds label to a graph edge when NGRAPH_ENABLE_VISUALIZE_TRACING=1
NGRAPH_VISUALIZE_TRACING_FORMAT | Default file format is .svg, one can change it to pdf or png use this
NGRAPH_VISUALIZE_TREE_OUTPUT_SHAPES | When enabled, add output shape of a node when NGRAPH_ENABLE_VISUALIZE_TRACING=1
NGRAPH_VISUALIZE_TREE_OUTPUT_TYPES | When enabled, add output type of a node when NGRAPH_ENABLE_VISUALIZE_TRACING=1

```